logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAvatarAssetUseCase.kt

```diff
@@ -35,17 +36,25 @@ interface GetAvatarAssetUseCase {
     suspend operator fun invoke(assetKey: UserAssetId): PublicAssetResult
 }
 
-internal class GetAvatarAssetUseCaseImpl(private val assetDataSource: AssetRepository) : GetAvatarAssetUseCase {
+internal class GetAvatarAssetUseCaseImpl(
+    private val assetDataSource: AssetRepository,
+    private val userRepository: UserRepository
+) : GetAvatarAssetUseCase {
     override suspend fun invoke(assetKey: UserAssetId): PublicAssetResult =
         // TODO(important!!): do local lookup for the profile pic before downloading a new one
         assetDataSource.downloadPublicAsset(assetKey.value, assetKey.domain).fold({
-            PublicAssetResult.Failure(it)
+            if (it.isNotFoundFailure) {
+                userRepository.removeUserBrokenAsset(assetKey)
+                PublicAssetResult.Failure(it, false)
+            } else {
+                PublicAssetResult.Failure(it, true)
+            }
         }) {
             PublicAssetResult.Success(it)
         }
 }
 
 sealed class PublicAssetResult {
     class Success(val assetPath: Path) : PublicAssetResult()
-    class Failure(val coreFailure: CoreFailure) : PublicAssetResult()
+    class Failure(val coreFailure: CoreFailure, val isRetryNeeded: Boolean) : PublicAssetResult()
```

Comment:
I don't see this `isRetryNeeded` being used here in kalium so will this `isRetryNeeded` be used in AR? 🤔